agent/job_runner.go

```diff
@@ -683,20 +683,22 @@ func (r *JobRunner) executePreBootstrapHook(hook string) (bool, error) {
 }
 
 // Starts the job in the Buildkite Agent API. We'll retry on connection-related
-// issues, but if a connection succeeds and we get an error response back from
+// issues, but if a connection succeeds and we get an client error response back from
 // Buildkite, we won't bother retrying. For example, a "no such host" will
-// retry, but a 422 from Buildkite won't.
+// retry, but an HTTP >= 400 <= 500 response from Buildkite won't.
 func (r *JobRunner) startJob(startedAt time.Time) error {
 	r.job.StartedAt = startedAt.UTC().Format(time.RFC3339Nano)
 
 	return roko.NewRetrier(
 		roko.WithMaxAttempts(30),
 		roko.WithStrategy(roko.Constant(5*time.Second)),
```

Comment:
If we're going to be retrying when we get 5xx errors, that might be when buildkite is experiencing an outage. we should probably change this to be a [`roko.Exponential`](https://pkg.go.dev/github.com/buildkite/roko#Exponential) strategy so that we don't accidentally nuke the buildkite backend when it's already having a bad day.

Reply:
Ah, good call. I've switched to the `roko.Exponential` strategy in the latest commit. Do you think 30 attempts with a 5 second exponential backoff is too much?

Reply:
i've done a bit of math, and if we did 30 attempts with a 5 second backoff, the the delay between try 29 and try 30 would be 29 Trillion Years, which feels maybe a bit too long 😅

I think 
```Go
		roko.WithMaxAttempts(7),
		roko.WithStrategy(roko.Exponential(2*time.Second, 0)),
```

which will wait 255s (~= 4min) total is a good balance - we wait a bit longer than the current constant delay, and try fewer times.

Reply:
😅 yes that's a bit much, I adjusted it downward as you suggested